monkey/monkey_island/cc/repository/i_agent_binary_repository.py

```diff
@@ -2,6 +2,10 @@
 from typing import BinaryIO
 
 
+class AgentRetrivalError(IOError):
```

Comment:
Use `RuntimeError`, not `IOError`. See https://github.com/guardicore/monkey/pull/1987#discussion_r891098724